src/Configuration.php

```diff
@@ -36,6 +39,17 @@ public function __construct(
          * Does the mapper should throw an exception if the target is read-only.
          */
         public bool $allowReadOnlyTargetToPopulate = false,
+        /**
+         * Classes with unknown properties, implemeting ArrayAccess.
+         *
+         * @var list<class-string<\ArrayAccess<string, mixed>>> $arrayAccessClasses
+         */
+        public array $arrayAccessClasses = [],
     ) {
+        $this->arrayAccessClasses[] = \ArrayObject::class;
+        if (class_exists(Document::class, false)) {
+            $this->arrayAccessClasses[] = Document::class;
+            $this->arrayAccessClasses[] = BSONDocument::class;
```

Comment:
`Document` is provided by an extension, so no need to call the class loader.
`BSONDocument` can be used only if the mongodb extension is loaded.